test/unit/test_ranges.py

```diff
@@ -0,0 +1,223 @@
+        account_path = os.path.join(self.swift_dir, 'account.ring.gz')
+        account_socket = listen(bind_address)
+        self.setup_ring(account_path, port=account_socket.getsockname()[1])
+        account_server = swift.account.server.AccountController(conf)
+
+        # Setup container server
+        container_path = os.path.join(self.swift_dir, 'container.ring.gz')
+        container_socket = listen(bind_address)
+        self.setup_ring(container_path, port=container_socket.getsockname()[1])
+        container_server = swift.container.server.ContainerController(conf)
+
+        # Setup object server
+        obj_ring_path = os.path.join(self.swift_dir, 'object.ring.gz')
+        obj_ring_socket = listen(bind_address)
+        self.setup_ring(obj_ring_path, port=obj_ring_socket.getsockname()[1])
+        object_server = swift_scality_backend.server.ObjectController(conf)
+
+        # Setup swift proxy
+        proxy_socket = listen(bind_address)
+        proxy = swift.proxy.server.Application(conf)
+
+        # Start swift proxy and auxiliary components.
+        spawn(wsgi.server, account_socket, account_server)
+        spawn(wsgi.server, container_socket, container_server)
+        spawn(wsgi.server, obj_ring_socket, object_server)
+        spawn(wsgi.server, proxy_socket, proxy)
+
+        # Create test account
+        account = 'a'
+        partition, acc_nodes = proxy.account_ring.get_nodes(account)
+        acc_node = acc_nodes.pop()
+
+        path = '/{device}/{partition}/{account}'.format(
+            device=acc_node['device'],
+            partition=partition,
+            account=account,
+        )
+        acc_headers = {'X-Timestamp': normalize_timestamp(time.time())}
+        acc_conn = HTTPConnection(acc_node['ip'], acc_node['port'])
+        acc_conn.request('PUT', path, headers=acc_headers)
+        r = acc_conn.getresponse()
+        self.assertEqual(r.status, 201)
+
+        # Create test container
+        self.proxy_host, self.proxy_port = proxy_socket.getsockname()
+        proxy_conn = HTTPConnection(self.proxy_host, self.proxy_port)
+        proxy_conn.request('PUT', '/v1/a/c/')
+        r = proxy_conn.getresponse()
+        self.assertEqual(r.status, 201)
+
+    def test_ranges(self):
+        self.check_path = '%s/a/c/o' % self.sproxyd_path
+        self.check_method = 'GET'
+        self.response = '_' * 100
+        object_path = '/v1/a/c/o'
+        proxy_conn = HTTPConnection(self.proxy_host, self.proxy_port)
+
+        # Get last 95 bytes
+        headers = {'RANGE': 'bytes=5-'}
+        self.expected_headers = {'RANGE': 'bytes=5-99'}
+        proxy_conn.request('GET', object_path, headers=headers)
+        r = proxy_conn.getresponse()
+        self.assertEqual(r.status, 206)
+        response_headers = dict(r.getheaders())
+        r.read()
+        self.assertEqual(response_headers['content-range'], 'bytes 5-99/100')
+
+        # Get last 5 bytes
+        headers = {'RANGE': 'bytes=-5'}
```

Comment:
There's some duplication of code here. Using a single test utility method and call it in several tests as mentioned before would remove this.